cachelib/navy/common/Device.h

```diff
@@ -111,14 +111,23 @@ class Device {
 
   // Write buffer to the device. This call takes ownership of the buffer
   // and de-allocates it by end of the call.
+  // @param placementHandle    Data placement identifier
   // @param buffer    Data to write to the device. It must be aligned the same
   //                  way as `makeIOBuffer` would return.
   // @param offset    Must be ioAlignmentSize_ aligned
-  bool write(uint64_t offset, Buffer buffer);
+  bool write(uint16_t placementHandle, uint64_t offset, Buffer buffer);
```

Comment:
When I used the term "placemnentHandle", I tried to cover any potential data placement technology; and it was not picked as an FDP term. I thought eventually it can be made as a class to abstract any data placement information.
If you want this arg as a last parameter, do you suggest two versions of write (with and without this argument) as in last code version?

Reply:
Yep, it is fair.

No, single version of write with `placementHandle` as the last param would be enough. Additionally, I would like to reserve `placementHandle` -1 or 0 (preferable) for non-FDP write.

Reply:
Also do you think using a class to abstract placementHandle would be a good idea?

Reply:
@arungeorge83 Sounds like we want to provide and use something like`HANDLE *`. Yeah, that makes sense, but I think that should be like [`FILE *`](https://cplusplus.com/reference/cstdio/FILE/) as opposed to [fd](https://man7.org/linux/man-pages/man2/open.2.html).

Reply:
@jaesoo-fb Actually I would like to have it like 'class RegionId'. A class to abstract the unit16_t id used for now. It also allows us to handle invalid/default cases easily.